### PR TITLE
[WIP] : MAUT-8990 : Upgrade to PHP8 

### PR DESCRIPTION
**Pull request by tejasync**
To be added...

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
         "wiki": "https://github.com/acquia/mc-cs-plugin-custom-objects/wiki"
     },
     "require": {
-        "php": ">=7.4",
```
Comment (escopecz) [inline composer.json]: I'm not sure we have to limit the max PHP version. This will prevent users to install it on PHP 8.2 and test that it works, possibly push fixes for it.

Comment (tejasync) [reply composer.json]: By keeping the existing option of `"php": ">=7.4` would state that the code will also work with PHP 9 and up. As there is no way of telling what backwards compatibility breaking changes will be introduced in PHP 9. 

IMO, specifying on what versions of PHP the code works would be good. If the users what to test the working for any other PHP versions, they should change it from here and then test it?

Instead of making such promises I'm thinking of making this as `"php": "^7.4 || ^8.0"`. This would indicate that following code would work on PHP 7.4.* and 8.0.*.

Comment (escopecz) [reply composer.json]: I was looking at some discussions on this topic and found only https://github.com/symfony/symfony/issues/42993#issuecomment-917719688

I understand from the comment that there are 2 lies we have to choose from:
1. We limit the PHP version with the upper boundary saying that it does not work with PHP 8.1 and above. Although we actually don't know that.
2. We leave the upper boundary open which may let people think that the plugin was tested with PHP 8.1 and 8.2 which is not true. But we allow them to use it with newer PHP versions if they want.

As Mautic 4 itself supports only PHP 7.4 and 8.0 at the moment then I think we don't have to worry about users using this plugin with PHP 8.1 and above. For M5 this plugin will need some refactoring where we can test it on 8.1 if Mautic 5 will support that at the time of release. But that's future problem.

In conclusion, I favor the second lie over the first one as it does not block users.

Comment (escopecz) [reply composer.json]: When I check for example Symfony library then it does not have the upper limit either https://github.com/symfony/symfony/blob/6.3/composer.json#L36

**Comment by escopecz** [comment]
Thanks Tejas! I guess we can close this now
